File: #1657.determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {

        vector<int> v1(26,0);
        for (int i = 0; i < word1.size(); i++) {
            v1[word1[i] - 'a']++;    
        }
        
        vector<int> v2(26,0);
        for (int i = 0; i < word2.size(); i++) {
            v2[word2[i] - 'a']++;
        }

        for (int i = 0; i < 26; i++) {
            if ((v1[i] > 0 && v2[i] == 0)
                || (v2[i] > 0 && v1[i] == 0)) {
                return false;
            }
        }

        std::sort(v1.begin(), v1.end());
        std::sort(v2.begin(), v2.end());
        return v1 == v2;
    }
};

File: #1657.determine-if-two-strings-are-close.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {boolean}
 */
var closeStrings = function(word1, word2) {
    if (word1.length !== word2.length) {
        return false;
    }

    let chars = [{}, {}];
    let notClose = false;
    for (let i = 0; i < word1.length; i ++) {
        if (!word1.includes(word2[i]) || !word2.includes(word1[i])) {
            notClose = true;
            break;
        }

        chars[0][word1[i]] ??= 0;
        chars[0][word1[i]] ++;
        chars[1][word2[i]] ??= 0;
        chars[1][word2[i]] ++;
    }

    if (notClose) {
        return false;
    }

    const values = Object.values(chars[1]);
    Object.values(chars[0]).forEach(n => {
        if (values.indexOf(n) < 0) {
            notClose = true;
            return;
        }

        values.splice(values.indexOf(n), 1);
    })

    return !notClose;
};
